Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/overview/ContactDiaryOverviewViewModel.kt

```diff
@@ -116,47 +166,95 @@ class ContactDiaryOverviewViewModel @AssistedInject constructor(
             drawableId = R.drawable.ic_low_risk_alert
         }
 
-        return DayOverviewItem.Risk(title, body, bodyExtend, drawableId)
+        return RiskEnfItem(title, body, bodyExtend, drawableId)
     }
 
-    private fun getEncountersForDate(
-        personEncounterList: List<ContactDiaryPersonEncounter>,
-        date: LocalDate
-    ) = personEncounterList
-        .filter { personEncounter -> personEncounter.date == date }
-        .map { personEncounter ->
-            DayOverviewItem.Data(
-                R.drawable.ic_contact_diary_person_item,
-                name = personEncounter.contactDiaryPerson.fullName,
-                duration = null,
-                attributes = getPersonAttributes(personEncounter),
-                circumstances = personEncounter.circumstances,
-                DayOverviewItem.Type.PERSON
-            )
+    private fun ContactDiaryPersonEncounter.toContactItemData(): ContactItem.Data = ContactItem.Data(
+        drawableId = R.drawable.ic_contact_diary_person_item,
+        name = contactDiaryPerson.fullName,
+        duration = null,
+        attributes = getPersonAttributes(this),
+        circumstances = circumstances,
+        type = ContactItem.Type.PERSON
+    )
+
+    private fun ContactDiaryLocationVisit.toContactItemData(): ContactItem.Data = ContactItem.Data(
+        drawableId = R.drawable.ic_contact_diary_location_item,
+        name = contactDiaryLocation.locationName,
+        duration = duration,
+        attributes = null,
+        circumstances = circumstances,
+        type = ContactItem.Type.LOCATION
+    )
+
+    private fun Map<ContactDiaryLocationVisit, TraceLocationCheckInRisk?>.toRiskEventItem(): RiskEventItem? {
+        if (isEmpty()) return null
+
+        val isHighRisk = values.any { it?.riskState == RiskState.INCREASED_RISK }
+
+        val body: Int = R.string.contact_diary_event_risk_body
+        val drawableID: Int
+        val title: Int
+
+        when (isHighRisk) {
+            true -> {
+                drawableID = R.drawable.ic_high_risk_alert
+                title = R.string.contact_diary_high_risk_title
+            }
+            false -> {
+                drawableID = R.drawable.ic_low_risk_alert
+                title = R.string.contact_diary_low_risk_title
+            }
```

Comment:
Could you please move these functions into a specific mapper instead of having them in the view model?

Reply:
We should refactor some things later in another PR.